device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -485,13 +500,35 @@ public void registerDeviceClients(Iterable<DeviceClient> deviceClients, long tim
                 log.info("Registering device {} to multiplexing client", configBeingRegistered.getDeviceId());
             }
 
-            this.deviceIO.registerMultiplexedDeviceClient(deviceClientConfigsToRegister, timeoutMilliseconds);
```

Comment:
Previously, this call would throw a MultiplexingClientDeviceRegistrationAuthenticationException even in partial success cases, which meant that the block of code below it that updates the local state of the multiplexedDeviceClients list never got updated to match. Now we'll catch the exception, update the local state, and then rethrow the exception